### PR TITLE
Fix/economy

### LINKED ISSUE #350: Buy Plots EconomyApi
### Issue description

- Expected result: What were you expecting to happen?
- Actual result: What actually happened?

If you buy a player's plot, the money will not be added to their account.
### Steps to reproduce the issue
1. ... Buy a Player Plot
2. ... Now you can see that the money has not been deposited into his account

### OS and versions
 1.9.0
* Plugin Version: Dev 653 Myplot
* PocketMine-MP: 3.15.2
* PHP: 7.3
* Game version: PE/Win10 

### Other Plugins
 Multiworld 1.5.2, EconomyApi 5.7.2, XpBank 1.0,
Worldprotect 3.0.0, EnchantUI 0.7

- If you remove all other plugins, does the issue still occur? Yes
- If the issue is **not** reproducible without other plugins:
  - Have you asked for help in the community discord before creating an issue? No
  - Can you provide sample, *minimal* reproducing code for the issue? If so, paste it in the bottom section.

### Console error, backtrace or other files
 There is no error.

### PR DESCRIPTION
**Pull request by DerDevHD**
## Introduction
This pull-request fixes the owner of the plot does not get his money

### Relevant issues

* Fixes #350 

### API changes
Added
```php
EconomyProvider::addMoney();
```

### Behavioural changes
/

## Backwards compatibility
/

## Follow-up
/

## Tests
Testes went successful

### CODE REVIEW

File: src/MyPlot/provider/EconomySProvider.php [hunk 1]
```diff
@@ -39,4 +38,21 @@ public function reduceMoney(Player $player, float $amount) : bool {
...
+	 * @param Player|string $player
+	 * @param float $amount
+	 * @return bool
+	 */
+	public function addMoney($player, float $amount) : bool {
+		if($amount < 1)
+			return true;
+		$ret = $this->plugin->addMoney($player, $amount, true, "MyPlot");
+		if($ret === EconomyAPI::RET_SUCCESS) {
+			$this->plugin->getLogger()->debug("MyPlot Reduced money of " . $player->getName());
```
Comment (jasonwynn10) [inline src/MyPlot/provider/EconomySProvider.php:52]: copy pasta

File: src/MyPlot/provider/EconomySProvider.php [hunk 2]
```diff
@@ -39,4 +38,21 @@ public function reduceMoney(Player $player, float $amount) : bool {
...
+			return true;
+		}
+		$this->plugin->getLogger()->debug("MyPlot failed to reduce money of ".$player->getName());
```
Comment (jasonwynn10) [inline src/MyPlot/provider/EconomySProvider.php:55]: copy  pasta

File: src/MyPlot/provider/EconomyProvider.php [hunk 3]
```diff
@@ -12,4 +12,12 @@ interface EconomyProvider {
...
 	 */
 	public function reduceMoney(Player $player, float $amount) : bool;
+
+	/**
+	 * @param Player|string $player
+	 * @param float $amount
+	 *
+	 * @return bool
+	 */
+	public function addMoney($player, float $amount) : bool;
```
Comment (jasonwynn10) [inline src/MyPlot/provider/EconomyProvider.php:22]: type hint `IPlayer` class here

**Review by jasonwynn10 — CHANGES_REQUESTED** [review]

File: src/MyPlot/provider/EconomyProvider.php (hunk 3 above)
Comment (DerDevHD) [reply src/MyPlot/provider/EconomyProvider.php:22]: okay

File: src/MyPlot/provider/EconomySProvider.php (hunk 2 above)
Comment (DerDevHD) [reply src/MyPlot/provider/EconomySProvider.php:55]: oops

File: src/MyPlot/provider/EconomySProvider.php (hunk 1 above)
Comment (DerDevHD) [reply src/MyPlot/provider/EconomySProvider.php:52]: oops

File: src/MyPlot/MyPlot.php [hunk 4]
```diff
@@ -898,6 +898,7 @@ public function buyPlot(Plot $plot, Player $player) : bool {
 		if($ev->isCancelled()) {
 			return false;
 		}
+		$this->getEconomyProvider()->addMoney($this->getServer()->getOfflinePlayer($plot->owner), $plot->price);
```
Comment (jasonwynn10) [inline src/MyPlot/MyPlot.php:901]: make sure to check if the payment fails. We don't want the plot transferred if the owner is never paid.

**Review by jasonwynn10 — CHANGES_REQUESTED** [review]

**Pull request merged**